### DISCUSSION TITLE
Need help: tell fuzzer not to send message

### DISCUSSION THREAD

**Discussion by poul1x** [jtpereyda/boofuzz#581] [Q&A] [answered]
I work with protocol, which can send response consisting of few packets. So, I want boofuzz not to send data after each receive. Just after "OK" is present in response message. Is it possible?

**Comment by SR4ven** [comment, marked as answer]
Right now two options to realize this come to my mind:
- Use `post_test_case_callbacks` in [session](https://boofuzz.readthedocs.io/en/stable/source/Session.html) to manually receive data and evaluate it until want to continue.
- Write your own connection class (or copy an existing one) with a custom receive function, which implements the desired feature.

**Comment by jtpereyda** [comment]
Those are both options -- `post_test_case_callbacks` strikes me as the easiest one.

This could be more easily supported if boofuzz had a more flexible message sequence setup -- something like a generic grammar or state machine rather than the request/response tree structure it has now.

**Comment by poul1x** [comment]
Ok, I will try this, thanks
